internal/services/labservice/lab_service_lab_plan_resource.go

```diff
@@ -0,0 +1,716 @@
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"disconnect_delay": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: azValidate.ISO8601Duration,
+					},
+
+					"idle_delay": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: azValidate.ISO8601Duration,
+					},
+
+					"no_connect_delay": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: azValidate.ISO8601Duration,
+					},
+
+					"shutdown_on_disconnect_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+
+					"shutdown_on_idle": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(labplan.ShutdownOnIdleModeUserAbsence),
+							string(labplan.ShutdownOnIdleModeLowUsage),
+							string(labplan.ShutdownOnIdleModeNone),
+						}, false),
+					},
+
+					"shutdown_enabled_when_not_connected": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+				},
+			},
+		},
+
+		"default_connection_profile": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"client_rdp_access": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(labplan.ConnectionTypeNone),
+							string(labplan.ConnectionTypePrivate),
+							string(labplan.ConnectionTypePublic),
+						}, false),
+					},
+
+					"client_ssh_access": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(labplan.ConnectionTypeNone),
+							string(labplan.ConnectionTypePrivate),
+							string(labplan.ConnectionTypePublic),
+						}, false),
+					},
+
+					"web_rdp_access": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(labplan.ConnectionTypeNone),
+							string(labplan.ConnectionTypePrivate),
+							string(labplan.ConnectionTypePublic),
+						}, false),
+					},
+
+					"web_ssh_access": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(labplan.ConnectionTypeNone),
+							string(labplan.ConnectionTypePrivate),
+							string(labplan.ConnectionTypePublic),
+						}, false),
+					},
+				},
+			},
+		},
+
+		"default_network_profile": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"subnet_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: networkValidate.SubnetID,
+					},
+				},
+			},
+		},
```

Comment:
Though this makes the schema easy to use at this moment, but later if the API added another sub-property "baz", then the schema has to introduce "foo_baz". Imagine if there are more such sub-properties introduced in API later, the schema will be a mess quickly. So we cannot do this.

Reply:
The implementation of this service is such that a VM Scale Set is provisioned, which has one network configuration, i.e. the default one, which from a Terraform perspective makes sense to be made top level properties. So in the case where an additional sub-property is added, the solution would be to add another top level property. 

Whilst this requires a little more work internally, it will provide a more consistent experience for users which is ultimately what we aim to achieve with the provider, for example like when we validate ID types for properties, or consistently choose to not expose `None`.

Hypothetically, if they were to introduce an additional network profile, we would opt to expose this either as a block or a separate resource depending on the behaviour of the API.

In this case this field needs to become a top level property.

Reply:
Updated to become a top level property.

Reply:
Updated